python/ray/dataframe/dataframe.py

```diff
@@ -2912,9 +2912,53 @@ def radd(self, other, axis='columns', level=None, fill_value=None):
 
     def rank(self, axis=0, method='average', numeric_only=None,
              na_option='keep', ascending=True, pct=False):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+
+        """
+        Compute numerical data ranks (1 through n) along axis.
+        Equal values are assigned a rank that is the [method] of
+        the ranks of those values.
+
+        Args:
+            axis (int): 0 or 'index' for row-wise,
+                        1 or 'columns' for column-wise
+            interpolation: {‘average’, ‘min’, ‘max’, ‘first’, ‘dense’}
+                Specifies which method to use for equal vals
+            numeric_only (boolean)
+                Include only float, int, boolean data.
+            na_option: {'keep', 'top', 'bottom'}
+                Specifies how to handle NA options
+            ascending (boolean):
+                Decedes ranking order
+            pct (boolean):
+                Computes percentage ranking of data
+        Returns:
+            A new DataFrame
+        """
+
+        def rank_helper(df):
+            return df.rank(axis=axis, method=method,
+                           numeric_only=numeric_only,
+                           na_option=na_option,
+                           ascending=ascending, pct=pct)
+
+        index = self.index
+
+        axis = pd.DataFrame()._get_axis_number(axis) if axis is not None \
```

Comment:
`axis=None` throws errors. Should not handle this case here.

```

In [3]: df.rank(axis=None)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-7bd95fb0ea7b> in <module>()
----> 1 df.rank(axis=None)

~/dev/anaconda3/envs/ray-dev/lib/python3.5/site-packages/pandas/core/generic.py in rank(self, axis, method, numeric_only, na_option, ascending, pct)
   5625         ranks : same type as caller
   5626         """
-> 5627         axis = self._get_axis_number(axis)
   5628 
   5629         if self.ndim > 2:

~/dev/anaconda3/envs/ray-dev/lib/python3.5/site-packages/pandas/core/generic.py in _get_axis_number(self, axis)
    355                 pass
    356         raise ValueError('No axis named {0} for object type {1}'
--> 357                          .format(axis, type(self)))
    358 
    359     def _get_axis_name(self, axis):

ValueError: No axis named None for object type <class 'pandas.core.frame.DataFrame'>```

Reply:
done